src/controller.cpp

```diff
@@ -228,23 +228,7 @@ void Controller::setPlan(
   const auto deltas = deltas_of_plan(global_plan_tf_);
```

Comment:
I think it makes more sense to combine these two functions.

` turning_radius_inv_vector_ = inverse_turning_radiuses(global_plan_tf_)`;

This reads better. That you need delta's is an implementation detail not relevent for this part of the code.

Reply:
I could do that but we also need the deltas for the distances to the goal (` distance_to_goal_vector_ = distances_to_goal(deltas);`). See PR #98 . If i merge the deltas into those functions, we'd have to do that work twice.

Reply:
Well, you can call that function as first step in this function right?

No duplicate code in that case.

Reply:
No duplicate code. But now `deltas_of_plan()` is called once in `Controller::setPlan()`:

```cpp
  const auto deltas = deltas_of_plan(global_plan_tf_);

  distance_to_goal_vector_ = distances_to_goal(deltas);
  turning_radius_inv_vector_ = inverse_turning_radiuses(deltas);
```

If we inline `deltas_of_plan()` in `distances_to_goal()` and `inverse_turning_radiuses()`, that work would get done twice. I'm not saying we shouldn't, it's just something to be aware of.

Reply:
Ah right, let's keep it like this!

Sorry, I was only looking at the delta :)